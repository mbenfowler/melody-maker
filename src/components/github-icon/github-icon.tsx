import React from 'react';
import { useStaticQuery, graphql } from 'gatsby';
import styled from 'styled-components';
import Img from 'gatsby-image';
import { motion } from 'framer-motion';

const Wrapper = styled.div``;

const GithubIcon: React.FC = () => {
  const data = useStaticQuery(graphql`
    query {
      icon: file(relativePath: { eq: "github-icon.png" }) {
        childImageSharp {
          fixed(height: 30, width: 30) {
            ...GatsbyImageSharpFixed_withWebp
          }
        }
      }
    }
  `);

  const imageData = data.icon.childImageSharp.fixed;
  return (
    <Wrapper>
      <motion.a
        href="https://github.com/mbfowler/melody-maker"
        rel="noopener noreferrer"
        target="_blank"
        css={`
          cursor: pointer;
        `}
        whileHover={{ opacity: 0.5 }}
      >
        <Img fixed={imageData} alt="GitHub Icon" />
      </motion.a>
    </Wrapper>
  );
};

export { GithubIcon };
