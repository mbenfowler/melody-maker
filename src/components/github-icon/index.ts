export { GithubIcon } from "./github-icon";
