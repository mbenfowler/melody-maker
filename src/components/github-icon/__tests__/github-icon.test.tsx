// TODO: Figure out why this test is failing.

import React from "react";
import { GithubIcon } from "../github-icon";
import { render, cleanup } from "@testing-library/react";
import * as Gatsby from "gatsby";
const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery");

beforeEach(() => {
  useStaticQuery.mockImplementationOnce(() => {
    return {
      icon: {
        childImageSharp: {
          fixed: "gatsby-logo.png",
        },
      },
    };
  });
});

afterEach(cleanup);

xit("matches snapshot", () => {
  const { asFragment } = render(<GithubIcon />);
  expect(asFragment()).toMatchSnapshot();
});
