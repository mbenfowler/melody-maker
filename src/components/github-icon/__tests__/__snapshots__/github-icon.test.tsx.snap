// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa bDWFJH"
  >
    <a
      href="https://github.com/gojutin/gatsby-starter-typescript-deluxe"
      rel="noopener noreferrer"
      target="_blank"
    >
      <div
        class=" gatsby-image-wrapper"
        style="position: relative; overflow: hidden; display: inline-block;"
      >
        <picture>
          <source />
          <img
            alt="GitHub Icon"
            loading="lazy"
            style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
          />
        </picture>
        <noscript>
          &lt;picture&gt;&lt;source srcset="undefined" /&gt;&lt;img loading="lazy" src="" alt="GitHub Icon" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
        </noscript>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
